.github/workflows/testing_and_building_repos.yml

```diff
@@ -0,0 +1,95 @@
+name: testing_and_building_repo
+on: [push, pull_request]
+jobs:
+  build: 
+    strategy:
+      matrix:
+        os: ['ubuntu-18.04']
+        perl: ['5.26'] 
+        
+    runs-on: ${{matrix.os}} 
+    name: Perl ${{ matrix.perl }} on ${{ matrix.os }}  
+    steps:
+    - uses: actions/checkout@v2 
+    
+    # Caching cpanm external modules
+    - name: Cache cpanm external modules
+      id: cpanmCache
+      uses: actions/cache@v2
+      with:
+        path: ~/perl5ext
+        key: ${{ matrix.os}}-build-cpanm-external
+
+    # Caching conda
+    - name: Cache conda
+      id: minicondaCache
+      uses: actions/cache@v2
+      with:
+        path: $HOME/miniconda
+        key: ${{ matrix.os}}-build-miniconda
+
+    #install libgd-dev and uuid-dev
+    - name: install libgd-dev and uuid-dev 
+      run:
+          sudo apt-get install -y libgd-dev uuid-dev 
+    
+    - name: install conda client, baton,samtools, set up conda environment
+      run: |
+          wget --quiet https://repo.anaconda.com/miniconda/Miniconda3-4.5.11-Linux-x86_64.sh -O miniconda.sh
```

Comment:
Please add any error messages to the PR, so we can see what's happening.

I expect the message suggesting a different version is their default "use the latest version" nag, which isn't always good advice because fresh releases (expecially major ones) can have more bugs that usual.

Working examplel using supplied Miniconda: https://github.com/wtsi-npg/perl-irods-wrap/blob/devel/.github/workflows/run-tests.yml

Reply:
`==> WARNING: A newer version of conda exists. <==
  current version: 4.9.2
  latest version: 4.10.1

Please update conda by running

    $ conda update -n base -c defaults conda


/home/runner/work/_temp/53425d3b-c88e-4376-be2b-641636f150b0.sh: line 16: activate: No such file or directory`
Above is the error I get when using the miniconda from the image. 
I'll try implementing the method used in `perl-irods-wrap`.

Reply:
That is caused by the way Actions uses a non-login shell by default, which means that `.bashrc` is ignored (which is where Conda puts its config, by default). The workaround writes some config to `.bash_profile` instead and sets the shells to be login by default.